python/sdk/docs/tutorial.rst

```diff
@@ -0,0 +1,666 @@
+          coordinator_1  | 2020-03-30T13:18:55.836805361+00:00   INFO xain_fl::coordinator::api: starting HTTP server on 0.0.0.0:8081
+          coordinator_1  | 2020-03-30T13:18:55.837086682+00:00   INFO warp::server: listening with custom incoming
+
+That's it, the platform is running! But before diving in, let's
+introduce the key concepts that power Federated Machine Learning.
+
+XAIN Federated Machine Learning in a nutshell
+---------------------------------------------
+
+.. note::
+
+   This section is a very quick introduction to the XAIN FL
+   Platform. A more in-depth description is available
+   `on XAIN's website <https://www.xain.io/federated-learning-technology>`_.
+
+Federated Machine Learning is a distributed machine learning
+approach. In its simplest form it is composed of a *coordinator* and a
+set of *participants*. The coordinator is responsible keeping any
+state required for the machine learning task, orchestrate the machine
+learning task across a set of participants, and perform the
+aggregation of the individual updates returned by the participants.
+
+In the XAIN FL Platform, the coordinator performs several rounds of
+training. For each round, it selects a subset of all the
+participants. These participants retrieve the latest global ML model
+from the coordinator, train on their local data, update the ML model
+locally, and finally send it back to the coordinator. Once all the
+participants selected by the coordinator have sent their results, the
+coordinator aggregate them which produces a new global ML model.
+
+
+.. _lifecycle:
+
+Participants lifecyle
+^^^^^^^^^^^^^^^^^^^^^
+
+In this tutorial, we're interested in writing *participants*. So let's
+take a closer look to a participant's lifecycle. When it starts, a
+participant should follow these steps:
+
+
+1. Connect to the coordinator
+2. Wait for being selected by the coordinator to take part to a round
+   of training
+3. Once selected, retrieve the latest training data from the
+   coordinator, in particular the model weights
+4. Train
+5. Send the training results to the coordinator, then go back to step
+   ``2.``
+
+With this knowledge, we're ready to start coding.
+
+Goal
+----
+
+To keep things simple, the participant we're going to implement won't
+solve a real machine learning problem. The idea is to write a
+minimalistic working example, that demonstrates that the system works.
+
+The model we'll use is a simple array with 5 identical ``float``
+values (for instance ``[1.2, 1.2, 1.2, 1.2, 1.2]``). At the beginning of round ``i``,
+let's suppose that the global model is ``[100.0, 100.0, 100.0, 100.0, 100.0]``. The participants that are selected to take part to round ``i`` will retrieve this model, pick a value between ``0`` and ``100.0``, and return an array with that value. For instance if a participant picks ``15.5``, it would send back to the coordinator an array filled with that value: ``[15.5, 15.5, 15.5, 15.5, 15.5]``.
```

Comment:
```suggestion
let's suppose that the global model is ``[100.0, 100.0, 100.0, 100.0, 100.0]``. The participants that are selected to take part in round ``i`` will retrieve this model, pick a value between ``0`` and ``100.0``, and return an array with that value. For instance if a participant picks ``15.5``, it would send back to the coordinator an array filled with that value: ``[15.5, 15.5, 15.5, 15.5, 15.5]``.
```